### ISSUE TITLE
Windows, cycle.serve does nothing

### ISSUE THREAD

**Issue by asktomsk** [greim/hoxy#53]
Hello,

Unfortunately, I can't figure out what is wrong with my code.
I don't see the content of `package.json` when I open `http://www.google.com/script.js`
But, It works good on MacOSx!

Windows 8.1
node v0.12.7
hoxy 3.0.0
``` javascript
var hoxy = require('hoxy');
var proxy = new hoxy.Proxy();
var port = 3080;

proxy.intercept({
    phase: 'request',
    fullUrl: 'http://www.google.com/script.js'
}, function (req, resp, cycle) {
    resp.statusCode = 200;
    console.log(__dirname + '\\package.json');
    return cycle.serve(__dirname + '\\package.json');

});

proxy.listen(port, 'localhost', function () {
    console.log('The proxy is listening on port ' + port + '.';);
});
```

**Comment by greim** [comment]
Are you seeing the output of `console.log(__dirname + '\\package.json')`?

**Comment by greim** [comment]
Another thing to check, is Strict Transport Security perhaps interfering here? In other words, does your browser automatically change `http` to `https` when you type in the URL, thus bypassing the proxy completely?

**Comment by asktomsk** [comment]
1 Yes, I see console output. So this method is calling 100%.
Also checked that Path separator is not the problem.

2 Browser doesn't redirect me to anything - it opens exact address.

It looks like `cycle.serve` does nothing.
content-length is 0

![image](https://cloud.githubusercontent.com/assets/1353194/8948714/11605fb8-3578-11e5-8f4a-e66fe85f1902.png)

I've added logging:
``` javascript
proxy.log('error warn', function (event) {
   console.error(event.level + ': ' + event.message);
   if (event.error) console.error(event.error.stack);
 });
```

And see this error in console:
```
Request
Z:\debugger\app/package.json
error: request phase error: connect EADDRNOTAVAIL
Error: connect EADDRNOTAVAIL
    at exports._errnoException (util.js:746:11)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1010:19)
```


------------------------------------------------------------
This code works good, as expected:

``` javascript
proxy.intercept({
    phase: 'request',
    fullUrl: 'http://www.google.com/script2.js',
    as: 'string'
}, function (req, resp, cycle) {
    console.log('Request');
    resp.string = 'OK';
    resp.statusCode = 200;
});

```

**Comment by asktomsk** [comment]
This code is also works:
``` javascript
proxy.intercept({
    phase: 'request',
    fullUrl: 'http://www.google.com/script3.js',
    as: 'buffer'
}, function (req, resp, cycle) {
    console.log('Request');
    resp.statusCode = 200;

    resp.buffer = fs.readFileSync(__dirname + '/package.json');
});
```

**Comment by greim** [comment]
Ah, okay. Seeing tests fail on Windows with similar errors. Reduced test case, this works on OS X but fails on Windows:

```js
'use strict';

let http = require('http');

let server = http.createServer(function(req, res) {
  res.end('hello');
}).listen(0);

let addr = server.address();

http.get({
  hostname: addr.address,
  port: addr.port,
  path: '/foo',
}, function(res) {
  console.log('hello');
  process.exit(0);
})
```

**Comment by greim** [comment]
Changing instances of `.listen(0)` to `.listen(0, 'localhost')` allows tests to pass. I just published 3.0.2. Let me know if that doesn't fix it.

**Comment by asktomsk** [comment]
The error message is gone, but the code still doesn't work.
It returns 404 to the browser.

I changed all my code to use `buffer` instead of `serve` and it works in any OS. But probably this isses needs to be figured out anyway.

**Comment by greim** [comment]
I see. Looks like a Windows pathing issue. Thanks for the catch.

**Comment by greim** [comment]
This might take a while to fix, because `serve` uses URL paths as filesystem paths, which works on Unix but fails on Windows. Complicating matters is that on Windows, URL paths are still expected to contain forward slashes, otherwise in some cases it causes "socket hang up". Hopefully will have something soon.

**Comment by greim** [comment]
Published 3.0.3, hopefully that fixes it.
